darts/preprocessing/base_transformer.py

```diff
@@ -0,0 +1,238 @@
+"""
+Base Transformer
+----------------
+"""
+from abc import ABC, abstractmethod
+from typing import TypeVar, Generic, Callable, Optional, List, Tuple
+
+from darts.logging import raise_if_not, get_logger
+from darts.preprocessing.validator import Validator
+
+logger = get_logger(__name__)
+T = TypeVar('T')
+
+
+class BaseTransformer(Generic[T], ABC):
+    def __init__(self,
+                 validators: Optional[List[Validator]] = None,
+                 validator_fns: Optional[List[Tuple[Callable[[T], bool], Optional[str]]]] = None,
+                 reversible: bool = False,
+                 can_predict: bool = False,
+                 fittable: bool = False):
+        """
+        Abstract class for transformers. All deriving classes have to implement only one function `transform`.
+        It also have `inverse_transform`, `fit` and `predict` left unimplemented. If child of this class implements
+        any of this methods it should mark this with appropriate property (`reversible`, `fittable`, `can_predict`).
+
+        Parameters
+        ----------
+        validators
+            List of validators that will be called before transform
+        validator_fns
+            List of tuples of validating function and optional string with reason for validating.
+            From this will be created list of validators and it will be appended to validators.
+        reversible
```

Comment:
EDIT: Just saw the code below to filter for fittable - maybe a dummy fit that doesn't change anything would be easier for users than remembering to pass this flag? Or can this flag be somehow automatically set if the fit function is overriden? I'm just worried about a scenario when someone implements fit() but forgets to set the right flag

Reply:
This is part I was spend quite some time to think about. Automatically detection if fit is overriden is possible - I didn't go with it because I felt like it was kinda tricky. I like this approach because, a) when user derives from this class and create custom transformer he has to explicitly mark what this transformer is capable of. b) if transformer is created from utility functions this properties are set automatically. But of course if you feel like setting this automatically is better we could go that way.

Reply:
I also think keeping those flags is fine, but I'd remove their default value. Then any sublcass calling `super().__init__` would have to specify the flags.